File: Java/GuessTheNumber/src/Guess.java
import java.util.Random;
import java.util.Scanner;

public class Guess {
    private int _randomNumber;
    boolean condition = true;
    boolean flag = true;
    Scanner sc = new Scanner(System.in);
    int numberOfTries;

    public void setRandomNumber(int number) {
        _randomNumber = number;
    }

    public int getRandomNumber() {
        return _randomNumber;
    }

    public void generateRandomNumber() {
        Random number = new Random();

        int numberGenerated = number.nextInt((10 - 1) + 1) + 1;
        setRandomNumber(numberGenerated);
    }

    public void enterNumber() {


        while (condition) {

            System.out.println("Enter your guess here: ");
            int number = sc.nextInt();
            numberOfTries += 1;

            if (number > getRandomNumber()) {
                System.out.println("Too high");
            } else if (number < getRandomNumber()) {
                System.out.println("Too Low");
            } else if (number == getRandomNumber()) {
                System.out.println("The number entered is CORRECT!!!");
                condition = false;
            } else {
                System.out.println("Please enter a number next time");
            }
        }

        System.out.println("You guessed the number in "  + numberOfTries + " tries.");

    }
}

File: C/GuessingGame/Guess.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>
#include <time.h>

int main() {
    // Guess a number from 0 - maxValue
    
    int maxValue = 5;
    int randomNumber;
    int incorrectGuess = true;
    int numberInput;

    srand(time(NULL));
    randomNumber = rand() % (maxValue + 1);
//    printf("%d", randomNumber);

    while (incorrectGuess)
    {
        printf("Guess the number between 0 - %d: \n", maxValue);
        scanf("%d", &numberInput);

        if (numberInput == randomNumber)
        {
            printf("You win!\n");
            incorrectGuess = false;
        } else if (numberInput > randomNumber) {
            printf("Too high!\n");
        } else if (numberInput < randomNumber) {
            printf("Too Low!\n");
        } else {
            printf("Enter a number next time!\n");
            break;
        }
        
    }
    printf("Thank you for playing!\n");

    return 0;
}
